core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -100,9 +100,15 @@ public int read(byte[] b) throws IOException {
 
   @Override
   public int read(byte[] b, int off, int len) throws IOException {
+    if (len == 0) {
+      return 0;
+    }
+    if (mPosition == mStatus.getLength()) { // at end of file
```

Comment:
`mPosition >= mStatus.getLength()`?

Reply:
how about updating `positionedRead` https://github.com/Alluxio/alluxio/pull/10749/files#diff-dda4e9db6a53b7319a9d5c19177847bcR173 ?

Reply:
Good catch, positioned read should be updated as well

Reply:
is `mPosition > mStatus.getLength()` possible?

Reply:
just to be safe and prevent hitting bugs elsewhere

Reply:
Done.